types/koa-session/index.d.ts

```diff
@@ -64,7 +64,7 @@ declare namespace session {
         /**
          * Hook: before save session
          */
-        beforeSave(...rest: any[]): void;
+        beforeSave?(...rest: any[]): void;
```

Comment:
https://github.com/koajs/session/blob/cb6d2c8448f21f6cf42efdb031798c9608f05602/lib/context.js#L215-L218

Should this be

```typescript
beforeSave?(ctx: Koa.Context, session: sessionProps): void;
```